test/js/utils-test.js

```diff
@@ -307,3 +311,65 @@ test("getRoles on element with abstract role.", function() {
         }
     }
 });
+
+test("getRoles on element with multiple valid roles.", function() {
+    var roles = ['checkbox', 'button', 'radio'];
+    var expectedRoles = roles.map(function(role){
+        var valid = (axs.constants.ARIA_ROLES.hasOwnProperty(role) && !axs.constants.ARIA_ROLES[role].abstract);
+        var result = { name: role, valid: valid };
+        if (valid) {
+            result.details = axs.constants.ARIA_ROLES[role];
+        }
+        return result;
+    });
+    var expected = {
+        valid: true,
+        applied: expectedRoles[0],
+        roles: expectedRoles
+    };
+    var element = document.createElement('div');
+    element.setAttribute('role', roles.join(' '));
+    var actual = axs.utils.getRoles(element);
+    deepEqual(actual, expected);
+});
+
+test("getRoles on element with invalid and valid roles.", function() {
+    var roles = ['foo', 'button', 'bar'];
+    var expectedRoles = roles.map(function(role){
```

Comment:
What do you think about scoping these two tests in an IIFE and pulling this function out into a named local helper function?